src/init.c

```diff
@@ -414,13 +448,26 @@ int read_cards(char *suggestion)
 				/* Get first count string */
```

Comment:
Good catch!

Proposal:
- while (true)
- hard fail, exit(1) if num_design gets higher than AVAILABLE_DESIGN

About num_design, I do not see where it is initialized to 0. implicit for a global variable? shall we add an explicit initialization. Or did I just miss something?

About retraining the nets, I agree it has to be done. if it's fine for you, I'll wait until we are confident that the rest of the code is ok. It takes really a long time.